### PR TITLE
Rebalances some things about progression traitors and adds some uplink items.

### PR DESCRIPTION
**Pull request by Watermelon914**
## About The Pull Request

Lowers TC rewards for assassination objectives, steal and destroy item objectives.
Makes it so that when you take the battlecruiser final objective, you become a battlecruiser ally and can assist with nuking the station. 
Moves the elite syndicate modsuit from the nukie uplink to the traitor uplink at a high cost and high progression cost.
Adds new uplink items that cause station-wide blackout and telecommunication disruption temporarily.

## Why It's Good For The Game

People can get an enormous amount of TC by completing these infinitely available objectives, and because of an oversight with the 'steal item' objective in regards to how much TC it rewards, they were worth a lot more TC than they should've been.
The battlecruiser objective changes have been made so that the traitor feels more involved with the battlecruiser they've summoned and can assist them with the nuke.
The new uplink items have been added so that there is more lategame gear that a traitor is able to purchase, as lategame traitor items are lacking right now. The elite syndicate modsuit gives traitors the opportunity to protect themselves during lategame engagements with security or other forces which could prove to be a threat when performing anything deeply antagonistic, like blowing up parts of the station in space.

## Changelog

:cl:
expansion: Final objective battlecruiser will now make you a battlecruiser ally, giving you the nuke codes and making it obvious to other battlecruiser members that you are one of them.
expansion: Added telecom disruption and blackout purchases for traitors to purchase, and moved the elite syndicate hardsuit from the nukie uplink to the traitor uplink as a high progression cost item.
balance: Rebalanced TC rewards from assassinations to be less, and lowered TC rewards for plentiful objectives.
balance: Lowers the progression reward of assassination objectives.
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
Does the progression traitor document have anything to say about what is an appropriate lategame item for traitors? I feel there could be a dangerous precedent set by giving traitors a nukie item, and I think there should be some further consideration for what is 'too much' and what is 'not enough' (if we consider things like deswords and scarp as not good enough currently to carry a lategame tot)

**Comment by Watermelon914** [comment]
> Does the progression traitor document have anything to say about what is an appropriate lategame item for traitors? I feel there could be a dangerous precedent set by giving traitors a nukie item, and I think there should be some further consideration for what is 'too much' and what is 'not enough' (if we consider things like deswords and scarp as not good enough currently to carry a lategame tot)

I will add a clarification to the document, ideally traitors should get stronger as time progresses, though deswords and such could be moved back to later times

**Comment by Watermelon914** [comment]
Here is the change in the design doc:
### TC Items
TC item progression cost should be balanced around the point at which you should be getting an item in a round. For example, deadly weapons such as deswords and pistols should be unlocked during midgame, where assassinations are unlocked so that assassination objectives can be more effectively performed.

If it's unclear when an item should be unlocked in an uplink, it should be balanced around the strength of the item, with stronger items taking longer to achieve.

**Comment by carshalash** [comment]
>Moves the elite syndicate modsuit from the nukie uplink to the traitor uplink at a high cost and high progression cost.

https://www.youtube.com/watch?v=H3elwDp1tC0

**Comment by Watermelon914** [comment]
> > Moves the elite syndicate modsuit from the nukie uplink to the traitor uplink at a high cost and high progression cost.
> 
> https://www.youtube.com/watch?v=H3elwDp1tC0

why?

**Comment by SkeletalElite** [comment]
While you're touching objective balance could you maybe make bugging head of staff rooms a little more vaiuable as they typically placed right in the middle of their departments making them very difficult to do stealthily and make bugging the captain's office a little less valuable as it's actually the easiest to bug without being noticed since it's typically in a secluded area and rarely actually has the captain in it and also has no windows.

**Comment by Mothblocks** [comment]
> People can get an enormous amount of TC by completing these infinitely available objectives

@Watermelon914 My only worry is that we *do* want traitors killing people, but maybe not encouraging huge chains of them. 

Would it be within scope to add a system where the rewards of objective types go down the more you do the same one? That is, in order to maximize TC, you would have to do a big swath of objectives? Do we not have enough for that?

If not, what you have here is probably fine, I just don't want traitors to feel like they should just be stealing a bunch of items and killing pets or something over what guarantees involvement and drama between another player.

File: code/modules/antagonists/traitor/objectives/destroy_item.dm [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 	progression_minimum = 20 MINUTES
 	progression_reward = 5 MINUTES
-	telecrystal_reward = list(2, 4)
+	telecrystal_reward = 0
```
Comment (Mothblocks) [inline code/modules/antagonists/traitor/objectives/destroy_item.dm:7]: This is only used as a subtype, right?

Comment (Watermelon914) [reply code/modules/antagonists/traitor/objectives/destroy_item.dm:7]: yes, most subtypes override it but it should be set to 0 by default

**Comment by Iamgoofball** [comment]
can we get this merged ASAP holy shitballs 15 crab-17s at once is not fun

**Comment by SinfulBliss** [comment]
Getting 0 TC for an objective, and like 2 TC for assassination feels like a scam. I had 3 assassination objectives on lowpop I chose to ignore because it felt wrong to effectively RR someone for a grand total of 3 TC (that's before the re-balance too).

**Comment by Watermelon914** [comment]
> Getting 0 TC for an objective, and like 2 TC for assassination feels like a scam. I had 3 assassination objectives on lowpop I chose to ignore because it felt wrong to effectively RR someone for a grand total of 3 TC (that's before the re-balance too).

it's not 0 TC, the TC amount changes depending on how long the objective takes. And also, assassination objectives are having their reward reduced because they're an 'infinite objective' compared to other objectives which are more limited.

**Pull request merged**